textile/fabric_pretreatment/doctype/pretreatment_order/pretreatment_order.py

```diff
@@ -237,7 +237,7 @@ def link_greige_fabric_in_ready_fabric(self):
 		if linked_greige_fabric != self.greige_fabric_item:
 			ready_fabric_doc = frappe.get_doc("Item", self.ready_fabric_item)
 			ready_fabric_doc.fabric_item = self.greige_fabric_item
-			ready_fabric_doc.save(ignore_permissions=True)
+			ready_fabric_doc.save(ignore_permissions=ignore_permissions)
```

Comment:
This should always be True. Check permissions separately before creating document